action/protocols/vote/protocol.go

```diff
@@ -7,18 +7,27 @@
 package vote
 
 import (
+	"fmt"
 	"math/big"
+	"sort"
 
 	"github.com/pkg/errors"
 
 	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocols/account"
 	"github.com/iotexproject/iotex-core/blockchain"
 	"github.com/iotexproject/iotex-core/iotxaddress"
+	"github.com/iotexproject/iotex-core/pkg/hash"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
 	"github.com/iotexproject/iotex-core/state"
 )
 
-// VoteSizeLimit is the maximum size of vote allowed
-const VoteSizeLimit = 278
+const (
+	// VoteSizeLimit is the maximum size of vote allowed
+	VoteSizeLimit = 278
+	// CandidatesPrefix is the prefix of the key of candidateList
+	CandidatesPrefix = "Candidates."
+)
 
 // Protocol defines the protocol of handling votes
 type Protocol struct{ bc blockchain.Blockchain }
```

Comment:
we don't need bc here. bc.StateByAddr() can be replaced by ws.State()

Reply:
Also i think we can put map[hash.PKHash]*state.Candidate into this struct, then we don't need pass the map around as input param below